src/components/LPolygon.vue

```diff
@@ -0,0 +1,41 @@
+<script>
+import { onMounted, ref, inject } from "vue";
+import { remapEvents, propsBinder } from "../utils.js";
+import { props, setup as polygonSetup } from "../functions/polygon";
+
+/**
+ * Polygon component, lets you add and customize polygon regions on the map
+ */
+export default {
+  name: "LPolygon",
+  props,
+  setup(props, context) {
+    const leafletRef = ref({});
+    const addLayer = inject("addLayer");
+
+    const { options, methods } = polygonSetup(props, leafletRef, context);
+
+    onMounted(async () => {
+      const { polygon, DomEvent, setOptions } = await import(
+        "leaflet/dist/leaflet-src.esm"
+      );
+
+      leafletRef.value = polygon(props.latLngs, options);
+
+      const listeners = remapEvents(context.attrs);
+      DomEvent.on(leafletRef.value, listeners);
+
+      propsBinder(methods, leafletRef.value, props, setOptions);
+
+      addLayer({
+        ...props,
+        ...methods,
+        leafletObject: leafletRef.value,
+      });
+    });
+  },
+  render() {
+    return null;
```

Comment:
**question:** since both Polygon and Rectangle are `layers` (since they inherit from polyline) does it make sense to have them render an invisible slot when something is added inside? (like polyline) WDYT?

Reply:
Great point, I agree! New commits pushed to add that.